lib/Parser/Parse.cpp

```diff
@@ -2020,14 +2020,29 @@ void Parser::EnsureStackAvailable()
 
 void Parser::ThrowNewTargetSyntaxErrForGlobalScope()
 {
-    //TODO: (falotfi) we need reliably distinguish eval in global scope vs in a function
-    // The rule for this syntax error is any time new.target is called at global scope
-    // we are excluding new.target in eval at global scope for now.
-    if(GetCurrentNonLamdaFunctionNode() == nullptr && (this->m_grfscr & fscrEvalCode) == 0)
+    if (GetCurrentNonLamdaFunctionNode() != nullptr)
     {
-        Error(ERRInvalidNewTarget);
+        return;
     }
-}
+
+    if ((this->m_grfscr & fscrEvalCode) != 0 && (this->m_grfscr & fscrIndirect) == 0)
+    {
+        Js::JavascriptFunction * caller = nullptr;
+        if (Js::JavascriptStackWalker::GetNonLamdaCaller(&caller, m_scriptContext))
```

Comment:
Exactly. Thanks.

-- Paul

From: suwc [mailto:notifications@github.com]
Sent: Wednesday, August 3, 2016 11:43 AM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Paul Leathers <pleath@microsoft.com>; Team mention <team_mention@noreply.github.com>
Subject: Re: [Microsoft/ChakraCore] Special cases of new.target (#1377)


In lib/Parser/Parse.cpp<https://github.com/Microsoft/ChakraCore/pull/1377#discussion_r73395674>:

>      }

> -}

> +

> +    if ((this->m_grfscr & fscrEvalCode) != 0 && (this->m_grfscr & fscrIndirect) == 0)

> +    {

> +        Js::JavascriptFunction * caller = nullptr;

> +        if (Js::JavascriptStackWalker::GetNonLamdaCaller(&caller, m_scriptContext))

Lamda => Lambda?
Will do.

________________________________

In reply to: 73384891<https://github.com/Microsoft/ChakraCore/pull/1377#discussion_r73384891>

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/1377/files/f89a104fd72f56eba5a59d03df8279f89162b642#r73395674>, or mute the thread<https://github.com/notifications/unsubscribe-auth/APF8ROruobVPRSzvWs2YSV62ni9DvM2sks5qcOExgaJpZM4Jb24l>.